x/hard/keeper/keeper.go

```diff
@@ -153,64 +154,76 @@ func (k Keeper) IterateBorrows(ctx sdk.Context, cb func(borrow types.Borrow) (st
 func (k Keeper) SetBorrowedCoins(ctx sdk.Context, borrowedCoins sdk.Coins) {
 	store := prefix.NewStore(ctx.KVStore(k.key), types.BorrowedCoinsPrefix)
 	if borrowedCoins.Empty() {
-		store.Set([]byte{}, []byte{})
+		store.Set(types.BorrowedCoinsPrefix, []byte{})
 	} else {
-		bz := k.cdc.MustMarshalBinaryBare(borrowedCoins)
-		store.Set([]byte{}, bz)
+		bz, err := borrowedCoins.MarshalJSON()
+		if err != nil {
+			panic(err)
+		}
+		store.Set(types.BorrowedCoinsPrefix, bz)
```

Comment:
I think it'd be slightly better to use protobuf here for coins marshalling.

The standard library json (un)marshalling is usually slow compared to gogo protobuf. And keeping store marshalling all proto makes it easier to reason about as a whole.

You could create a wrapper type around `sdk.Coins` a bit like `sdk.DecProto` or `cdptypes.OwnerCDPIndex` and use it only for (un)marshalling to the store.